pkgs/development/compilers/oraclejdk/jdk-linux-base.nix

```diff
@@ -67,19 +77,23 @@ let
 
 in
 
+assert architectures ? ${stdenv.hostPlatform.system};
```

Comment:
can we just check whether `${stdenv.hostPlatform.system}` appears in `sha256`, instead of also introducing `architectures`?